### PR TITLE
Fix boundary extraction from Content-Type with additional parameters

### PR DESCRIPTION
**Pull request by iliaal**
### Motivation

`MultipartFormDataParser::parse()` extracts the boundary with a greedy pattern:

```php
preg_match('/boundary="?(.*)"?$/is', $contentType, $matches)
```

For a header like `multipart/form-data; boundary=ABC; charset=utf-8` it captures `ABC; charset=utf-8` as the boundary, `preg_split` finds no delimiters and parsing returns no parts — real-world servers and clients do append additional Content-Type parameters. Quoted boundaries with trailing parameters (`boundary="ABC"; charset=utf-8`) fail the same way.

### Changes

Match either a quoted boundary string or an unquoted token terminated by the next parameter separator or whitespace (per the MIME parameter syntax), and bail out on an empty boundary:

```php
preg_match('/boundary=(?:"([^"]*)"|([^;\s]*))/is', $contentType, $matches)
```

Existing behavior for plain and quoted boundaries without trailing parameters is unchanged.

### Tests

New cases in `tests/framework/web/MultipartFormDataParserTest.php`: boundary followed by `; charset=utf-8` (unquoted and quoted forms) parses all parts; missing/empty boundary returns no parts.

### CODE REVIEW

**Review by terabytesoftw — COMMENTED** [review]
Thanks for the fix. Additionally, please add the corresponding entry to `framework/CHANGELOG.md`.

File: framework/web/MultipartFormDataParser.php [hunk 1]
```diff
@@ -142,11 +142,14 @@ public function parse($rawBody, $contentType)
             return [];
         }
 
-        if (!preg_match('/boundary="?(.*)"?$/is', $contentType, $matches)) {
+        if (!preg_match('/boundary=(?:"([^"]*)"|([^;\s]*))/is', $contentType, $matches)) {
             return [];
         }
 
-        $boundary = trim($matches[1], '"');
+        $boundary = isset($matches[1]) && $matches[1] !== '' ? $matches[1] : $matches[2];
```
Comment (terabytesoftw) [inline framework/web/MultipartFormDataParser.php]: An empty quoted boundary (`boundary=""`) leaves `$matches[2]` unset. Because `$matches[1]` is empty, this expression accesses a missing array key; Yii converts the warning into an `ErrorException`, which can turn malformed input into an HTTP 500. Please use a safe fallback (or require non-empty captures) and add a test for the quoted-empty form.

File: framework/web/MultipartFormDataParser.php [hunk 2]
```diff
@@ -142,11 +142,14 @@ public function parse($rawBody, $contentType)
             return [];
         }
 
-        if (!preg_match('/boundary="?(.*)"?$/is', $contentType, $matches)) {
+        if (!preg_match('/boundary=(?:"([^"]*)"|([^;\s]*))/is', $contentType, $matches)) {
```
Comment (terabytesoftw) [inline framework/web/MultipartFormDataParser.php]: This pattern searches for the substring `boundary=` rather than a real parameter name. For example, `x-boundary=wrong; boundary=real` selects `wrong`, and text inside another quoted parameter can also be selected. Please parse parameter boundaries while respecting quoted strings and add regression coverage. Merely anchoring after `;` is insufficient when the semicolon occurs inside a quoted value.

**Review by terabytesoftw — COMMENTED** [review]
Please address the following correctness issues before approval.

File: framework/web/MultipartFormDataParser.php (hunk 1 above)
Comment (iliaal) [reply framework/web/MultipartFormDataParser.php]: Verified: `boundary=""` leaves `$matches[2]` unset (PHP 8.4 warning, Yii ErrorException). Parser now returns the empty value and bails out. Tests cover `boundary=""` and `boundary=""; charset=utf-8`.

File: framework/web/MultipartFormDataParser.php (hunk 2 above)
Comment (iliaal) [reply framework/web/MultipartFormDataParser.php]: Verified: `x-boundary=wrong; boundary=real` selected `wrong`, and `;` / `boundary=` inside another quoted parameter did too. Parameters are now split only outside quoted-strings. Those cases are in `testParseWithBoundaryFollowedByAdditionalParameters`.

**Comment by iliaal** [comment]
Added under 2.0.56.

**Review by terabytesoftw — APPROVED** [review]

**Review by Arhell — COMMENTED** [review]
Lgtm
